<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Reilly Brennan" />


<title>About</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Spillover risk in a host-ectoparasite network</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">About</a>
</li>
<li>
  <a href="Maps.html">Rodent Richness Maps</a>
</li>
<li>
  <a href="Phylogeny.html">Rodent Phylogeny</a>
</li>
<li>
  <a href="GeoProb.html">Ectoparasite Sharing: Geographic Range</a>
</li>
<li>
  <a href="GenProb.html">Ectoparasite Sharing: Phylogenetic Relatedness</a>
</li>
<li>
  <a href="Network.html">Rodent-Ectoparasite Network</a>
</li>
<li>
  <a href="Mantel.html">Network Analysis: Mantel Tests</a>
</li>
<li>
  <a href="Closcen.html">Network Analysis: Closeness Centrality</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">About</h1>
<h4 class="author">Reilly Brennan</h4>

</div>


<hr />
<div id="objective" class="section level3">
<h3>Objective</h3>
<p>To determine how geographic, genetic, and ecological interactions
between rodents in a network can be used to explore the risk of disease
spillover.</p>
</div>
<div id="the-system" class="section level3">
<h3>The System</h3>
<p>We are focusing on a rodent-ectoparasite network in the neotropics
(Chile), where hantavirus in endemic. There are 50 rodent species
documented to be parasitized by a total of 131 different ectoparasites
which include fleas, ticks, mites and lice. We updated a review of
Chilean rodents and ectoparasites <span
class="citation">(Landaeta-Aqueveque et al. 2021)</span> to include
associations documented until March 2023.</p>
</div>
<div id="methods" class="section level3">
<h3>Methods</h3>
<p>We use host-ectoparasite associations to build an ecological network
of rodent hosts where they are connected by ectoparasites they share.
Based on this, we explore the probability for rodents to share
ectoparasites based on range overlap, range distance, and phylogenetic
relatedness. We then assess how well the network model predicts
Hantavirus hosts and use it to assess potential novel hosts.</p>
<p><strong>Maps</strong></p>
<ul>
<li>Explore the rodent-ectoparasite network by first creating rodent
richness maps for all rodent species in Chile and only rodents that are
documented with ectoparasites.</li>
<li>Create a range overlap and range distance matrix.</li>
</ul>
<p><strong>Phylogeny</strong></p>
<ul>
<li>Assess the phylogenetic relationships between rodent species in the
dataset</li>
<li>Build a phylogeny using Geneious Prime software and export the
distance matrix.</li>
</ul>
<p><strong>Ectoparasite sharing</strong></p>
<ul>
<li>Use the GeoTax package <span class="citation">(Robles 2023)</span>
to calculate the probability for rodents to share ectoparasites based on
range overlap, range distance, and phylogenetic relatedness.</li>
<li>Make plot showing the probability to share ectoparasites for each
ectoparasite and for the whole dataset.</li>
</ul>
<p><strong>Network building</strong></p>
<ul>
<li>Use an adjacency matrix to build a network in the igraph
package.</li>
<li>Use the igraph package to find the shortest paths between all pairs
of rodents in the network.</li>
</ul>
<p><strong>Network Analysis</strong></p>
<ul>
<li>Test for correlations between network distances with phylogenetic
and range distance and between range and phylogenetic distance using
mantel tests.</li>
<li>Use the closeness centrality from Gephi to split the rodents into
quantiles and visually plot them based on their groups.</li>
<li>Test how well closeness centrality predicts hantavirus hosts using a
binomial logistic regression.</li>
</ul>
<p>In addition to the code documented on this site, we used Gephi
software <span class="citation">(Bastian M., n.d.)</span> to build our
rodent-ectoparasite network and to calculate scores of closeness
centrality for each rodent. We also used Geneious Prime 2023.0.4 (<a
href="https://www.geneious.com"
class="uri">https://www.geneious.com</a>), to build a neighbor-joining
phylogeny based on MUSCLE alignment to caluclate pairwise phylogenetic
distance between rodent species. We used cytpchrome b sequences
downloaded from GenBank <span class="citation">(Sayers et al.
2022)</span>.</p>
</div>
<div id="network-model" class="section level3">
<h3>Network Model</h3>
<p>This network shows rodents as nodes (cirles) and shared ectoparasites
as edges (connecting lines). The edges are weighted by how many
ectoparasites the rodents share and the nodes are weighted by the
closeness centrality of the rodent in the network. The colors correspond
to the closeness centrality quantile group that the rodent belongs to
with yellow being the most central and dark blue being the least. We
used Gephi software to construct this model. <img
src="NetworkColors.png" /></p>
<p>Cited:</p>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-RN109" class="csl-entry">
Bastian M., Jacomy M., Heymann S. n.d. <span>“Gephi: An Open Source
Software for Exploring and Manipulating Networks.”</span> Conference
Proceedings. In <em>International AAAI Conference on Weblogs and Social
Media</em>.
</div>
<div id="ref-RN100" class="csl-entry">
Landaeta-Aqueveque, C., L. Moreno Salas, A. Henríquez, M. C. Silva-de la
Fuente, and D. González-Acuña. 2021. <span>“Parasites of Native and
Invasive Rodents in Chile: Ecological and Human Health Needs.”</span>
Journal Article. <em>Front Vet Sci</em> 8: 643742. <a
href="https://doi.org/10.3389/fvets.2021.643742">https://doi.org/10.3389/fvets.2021.643742</a>.
</div>
<div id="ref-RN114" class="csl-entry">
Robles, Angel. 2023. <span>“Geotax: Tools for Exploring Taxonomic
Distance over the Geographical Space.”</span> Computer Program.
</div>
<div id="ref-RN110" class="csl-entry">
Sayers, E. W., E. E. Bolton, J. R. Brister, K. Canese, J. Chan, D. C.
Comeau, R. Connor, et al. 2022. <span>“Database Resources of the
National Center for Biotechnology Information.”</span> Journal Article.
<em>Nucleic Acids Res</em> 50 (D1): D20–d26. <a
href="https://doi.org/10.1093/nar/gkab1112">https://doi.org/10.1093/nar/gkab1112</a>.
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
